src/dialogue.h

```diff
@@ -256,11 +248,10 @@ struct dialogue {
         std::vector<mission *> missions_assigned;
 
         talk_topic opt( dialogue_window &d_win, const std::string &npc_name, const talk_topic &topic );
-
         dialogue() = default;
-        talker *actor( const bool is_beta ) const {
-            return ( is_beta ? beta : alpha ).get();
-        }
+        explicit dialogue( std::unique_ptr<talker> alpha_in,
+                           std::unique_ptr<talker> beta_in );
+        talker *actor( const bool is_beta ) const;
```

Comment:
Might want to consider using an enum or something instead of raw bools. Or a `beta/alpha_if_exists()` or similar function.

Reply:
I don't follow quite what you mean here.  Like an enum with values alpha_only,beta_only and both?  I'm not sure that is any more clear or easy to follow.

I did try this originally with cata::optional but couldn't get that to work.

Reply:
Just `alpha` or `beta`. It's more clear than `true` or `false`.
e.g. if I come across `d.actor( false )->can_see()`, I need to look up what the argument means, instead of just knowing with `d.actor( alpha )->can_see()`.